File: leetcode/python/Same Tree/Same_Tree.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param p, a tree node
    # @param q, a tree node
    # @return a boolean
    def isSameTree(self, p, q):
        if p is None and q is None :
            return True
        if p is not None and q is not None:
            return p.val == q.val and self.isSameTree(q.left,p.left) and self.isSameTree(q.right,p.right)
        return False

File: leetcode/cpp/SameTree.cpp
//
// Created by zhangqi on 2018/1/5.
//Time : O(n)
//Space : O(h)

#include "TreeNode.h"

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(!p && !q){
            return true;
        }
        return p && q && p->val == q->val && isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
    }
};